mycore-solr/src/main/java/org/mycore/solr/MCRSolrClientFactory.java

```diff
@@ -124,19 +120,30 @@ public static void setSolrClient(String serverURL) {
         add(baseCore);
         DEFAULT_CORE_NAME = defaultCore.getName();
     }
+    **/
 
     /**
-     * Sets the new solr url.
+     * Sets the new solr server url and core name as default client.
      * 
      * @param serverURL base solr url
      * @param core core of the server
      */
-    public static void setSolrClient(String serverURL, String core) {
+    public static void setSolrDefaultClient(String serverURL, String core) {
         removeDefaultCore();
         add(new MCRSolrCore(serverURL, core));
         DEFAULT_CORE_NAME = core;
     }
 
+    /**
+     * Sets the new solr server url and core name.
+     * 
+     * @param serverURL base solr url
+     * @param core core of the server
+     */
+    public static void setSolrClient(String serverURL, String core) {
```

Comment:
This method should be renamed to  something like add or addCore. The name setSolrClient indicates there is only one solr client in this Factory and you overwrite it, but it its only a shorthand to add a new solr core.